0000-node-requirements/0000-node-requirements.md

```diff
@@ -0,0 +1,150 @@
+---
+title: Interledger Node - Requirements Specification
+draft: 1
+---
+# Interledger Node - Requirements Specification
+
+This document defines the basic functions of an ILP node. 
+
+An ILP node is a system that performs the necessary functions to route ILP packets between peers on the open Interledger network.
+
+The document is specific to the requirements of a "core" node that only operates at the ILP layer and does not concern itself with functionality in the higher protocol layers (e.g. STREAM, PSK2, etc). In other words, this document describes the functions of a node that is neither a sender or receiver of ILP packets but rather a "middlebox", to borrow some Internet terminology.
+
+Such a node is often referred to as a **connector** as it provides connectivity between different networks.
+
+# Overview
+
+An ILP node is very similar to a network router on an IP network. It has multiple incoming and outgoing links on which it sends and receives packets of data. Its primary function is to route incoming packets, on any links, to the appropriate outgoing link.
+
+Where an ILP node differs from an IP router is that it MUST be stateful. ILP packets come in request/response pairs so a node that receives a request on one link, and then routes it out on another, MUST ensure it is able to match any response it gets back on the outgoing link to the original request sent on that link. It MUST also route the response back along the same incoming link on which it received the original incoming request.
+
+The business logic of an ILP node is also more sophisticated than an IP router, in that the node operator is paid, per packet, by the sender of incoming packets to perform this function. It follows therefore, that the operator must also pay the next node for every packet it sends to it.
+
+However, payment for forwarding a packet is only due if a **valid** response to a request is received **within a predefined expiry** therefor the node must also track the payments owed and due for accepting and forwarding packets, must reconcile the amounts owed to, and due from, peers and respond to (or initiate) external settlement events on those accounts.
```

Comment:
therefor -> therefore

Reply:
Actually, that whole sentence doesn't really roll off the old tongue eh ?

Reply:
Broke up into smaller sentences